cmasher/tests/test_utils.py

```diff
@@ -65,6 +66,113 @@ def _MPL38_colormap_eq(cmap, other) -> bool:
+        "cmaps, nodes",
+        [
+            (["Blues", "Oranges", "Greens"], [0.25, 0.75]),
+            (["Blues", "Oranges", "Greens"], np.array([0.25, 0.75])),
+            (
+                [
+                    mpl.colormaps["Blues"],
+                    mpl.colormaps["Oranges"],
+                    mpl.colormaps["Greens"],
+                ],
+                [0.25, 0.75],
+            ),
+        ],
+    )
+    def test_combine_cmaps(self, cmaps, nodes):
+        combined_cmap = combine_cmaps(*cmaps, nodes=nodes, n_rgb_levels=256)
+        blues_cmap = mpl.colormaps["Blues"]
+        oranges_cmap = mpl.colormaps["Oranges"]
+        greens_cmap = mpl.colormaps["Greens"]
+
+        assert np.allclose(combined_cmap(0.0), blues_cmap(0))
+        assert np.allclose(combined_cmap(0.25), oranges_cmap(0))
+        assert np.allclose(combined_cmap(0.75), greens_cmap(0))
+        assert np.allclose(combined_cmap(1.0), greens_cmap(255))
+
+        assert combined_cmap.N == 256
+
+    # Test if combining less than 2 colormaps triggers an error
+    @pytest.mark.parametrize(
+        "cmaps",
+        [
+            pytest.param([], id="no_cmap"),
+            pytest.param(["Blues"], id="single_cmap"),
+            pytest.param(["fake_name"], id="fake_cmap_name"),
+        ],
+    )
+    def test_not_enough_cmaps(self, cmaps):
+        with pytest.raises(
+            ValueError, match="Expected at least two colormaps to combine."
+        ):
+            combine_cmaps(*cmaps)
+
+    # Test if invalid colormap name raise an error
+    def test_invalid_cmap_name(self):
+        with pytest.raises(
+            KeyError,
+            match="'fake_cmap' is not a known colormap name",
+        ):
+            combine_cmaps("fake_cmap", "Blues")
+
+    # Test if invalid colormap types raise an error
+    @pytest.mark.parametrize(
+        "invalid_cmap",
+        [0, 0.0, [], ()],
+    )
+    def test_invalid_cmap_types(self, invalid_cmap):
+        with pytest.raises(
+            TypeError,
+            match=f"Unsupported colormap type: {type(invalid_cmap)}.",
+        ):
+            combine_cmaps("Blues", invalid_cmap)
+
+    # Test if invalid nodes types raise an error
+    @pytest.mark.parametrize("invalid_nodes", [0.5])
+    def test_invalid_nodes_types(self, invalid_nodes):
+        with pytest.raises(
+            TypeError,
+            match=f"Unsupported nodes type: {type(invalid_nodes)}.",
+        ):
+            combine_cmaps("Blues", "Greens", nodes=invalid_nodes)
+
+    # Test if mismatch cmaps and nodes length raise an error
+    @pytest.mark.parametrize(
+        "cmaps, nodes",
+        [
+            (["Blues", "Oranges", "Greens"], [0.5]),
+            (["Reds", "Blues"], [0.2, 0.8]),
+        ],
+    )
+    def test_cmaps_nodes_length_mismatch(self, cmaps, nodes):
+        with pytest.raises(
+            ValueError,
+            match="Number of nodes should be one less than the number of colormaps.",
```

Comment:
It seems I'm having some problem wrapping this line (`line too long (85 > 79 characters)Flake8(E501)`), if I change it to:
```python
def test_cmaps_nodes_length_mismatch(self, cmaps, nodes):
    with pytest.raises(
        ValueError,
        match=("Number of nodes should be one less than "
               "the number of colormaps."
           ),
    ):
      combine_cmaps(*cmaps, nodes=nodes)
```
`ruff-format` would then complain and change it to:
```python
def test_cmaps_nodes_length_mismatch(self, cmaps, nodes):
    with pytest.raises(
        ValueError,
        match=(
            "Number of nodes should be one less than " "the number of colormaps."
        ),
    ):
        combine_cmaps(*cmaps, nodes=nodes)
```

Reply:
It looks like this isn't a problem anymore, is it ?

Reply:
Sorry I still haven't resolve this issue... For me the formatting requirement of `ruff-format` and `flake8` seems to go against each other, `flake8` would complain about line too long if I put the message `str` in a single line, but if I wrap it into two lines `ruff-format` would then put it back to a single liner... I'm still drafting the docs and I would try to resolve this soon.